### PR TITLE
Added intercom endpoints to save account and customer data

### PR DESCRIPTION
**Pull request by ruthwickp**
Provided 3 separate endpoints to download the accounts, users, and membership information. Endpoints can only be hit by specific users.

### CODE REVIEW

File: app/controllers/intercom_integrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  before_action :authenticate_user!
+
+  def load_accounts
+    company, intercom = extract_company_and_intercom_client
+    return unless intercom
+
+    Rails.logger.info("Loading intercom accounts for company #{company.name}...")
+    accounts = company.accounts
+    account_names = accounts.map(&:name).to_set
+    companies_to_add = intercom.companies.scroll.reject { |c| account_names.include?(c.name) }
```
Comment (ruthwickp) [inline app/controllers/intercom_integrations_controller.rb]: Why reject?

File: Gemfile [hunk 2]
```diff
@@ -34,7 +35,7 @@ gem 'omniauth-intercom', '0.1.9'
 gem "pg", "0.21.0"
 gem "php-serialize", '1.2.0'
 gem 'postmark-rails', '0.21.0'
-gem 'puma', '3.12.4'
+gem 'puma', git: 'https://github.com/eric-norcross/puma.git', branch: 'chrome_70_ssl_curve_compatiblity'
```
Comment (ruthwickp) [inline Gemfile]: Separate this out in development and production

**Comment by ruthwickp** [comment]
* Added migration to save the source and source identifiers
* Upgraded puma and capistrano puma to handle https

**Comment by ruthwickp** [comment]
Let's wait until we hear back from Curate before merging this.

File: app/controllers/intercom_integrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    [company, intercom]
+  end
+
+  def convert_to_account_and_save(intercom_company)
+    a = current_company.accounts.new
+    a.name = intercom_company.name
+    a.domain = intercom_company.name
+    a.mrr = intercom_company.monthly_spend
+    a.paying = a.mrr.present? && a.mrr.positive?
+    a.source = 'Intercom'
```
Comment (pranav7) [inline app/controllers/intercom_integrations_controller.rb:72]: Might make sense to create a constants file for valid sources, for example: https://github.com/pranav7/cadet-app/blob/2de42af4a4736132a8445597ad6822a9d97e98cb/app/lib/constants/event_types.rb#L1-L10

File: app/controllers/intercom_integrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  def extract_company_and_intercom_client
+    company = Company.find(params[:company_id])
+    intercom_access_token = company.company_setting.intercom_access_token
+    intercom = nil
+    intercom = Intercom::Client.new(token: intercom_access_token) if intercom_access_token
+    [company, intercom]
+  end
+
+  def convert_to_account_and_save(intercom_company)
+    a = current_company.accounts.new
```
Comment (pranav7) [inline app/controllers/intercom_integrations_controller.rb:67]: Forgive me for shouting code styling 🙈, but nice to name variables for readability 

```suggestion
    account = current_company.accounts.new
```

**Pull request closed without merging**